nemo/collections/nlp/models/language_modeling/megatron/tokens_encoder_decoder.py

```diff
@@ -0,0 +1,282 @@
+            use_cpu_initialization=use_cpu_initialization,
+            embedding_dropout_prob=hidden_dropout,
+        )
+        self.decoder_embedding = self.encoder_embedding
+        self._encoder_embedding_key = "encoder_embedding"
+        self._decoder_embedding_key = "decoder_embedding"
+
+        encoder = get_encoder_model(
+            arch=encoder_arch,
+            hidden_size=hidden_size,
+            ffn_hidden_size=hidden_size,
+            num_layers=num_layers,
+            num_attention_heads=num_attention_heads,
+            apply_query_key_layer_scaling=apply_query_key_layer_scaling,
+            kv_channels=kv_channels,
+            init_method=init_method_normal(init_method_std),
+            scaled_init_method=scaled_init_method_normal(init_method_std, num_layers),
+            encoder_attn_mask_type=AttnMaskType.padding,
+            pre_process=pre_process,
+            post_process=post_process,
+            init_method_std=init_method_std,
+            use_cpu_initialization=init_method_std,
+            hidden_dropout=hidden_dropout,
+            precision=precision,
+            fp32_residual_connection=fp32_residual_connection,
+            activations_checkpoint_method=activations_checkpoint_method,
+            activations_checkpoint_num_layers=activations_checkpoint_num_layers,
+            layernorm_epsilon=layernorm_epsilon,
+            bias_gelu_fusion=bias_gelu_fusion,
+            persist_layer_norm=persist_layer_norm,
+            openai_gelu=openai_gelu,
+            onnx_safe=onnx_safe,
+            hidden_steps=hidden_steps,
+            hidden_blocks=hidden_steps,
+        )
+
+        decoder = get_decoder_model(
+            arch=decoder_arch,
+            hidden_size=hidden_size,
+            ffn_hidden_size=hidden_size,
+            num_layers=num_layers,
+            num_attention_heads=num_attention_heads,
+            apply_query_key_layer_scaling=apply_query_key_layer_scaling,
+            kv_channels=kv_channels,
+            init_method=init_method_normal(init_method_std),
+            scaled_init_method=scaled_init_method_normal(init_method_std, num_layers),
+            decoder_attn_mask_type=AttnMaskType.causal,
+            pre_process=pre_process,
+            post_process=post_process,
+            init_method_std=init_method_std,
+            use_cpu_initialization=init_method_std,
+            hidden_dropout=hidden_dropout,
+            precision=precision,
+            fp32_residual_connection=fp32_residual_connection,
+            activations_checkpoint_method=activations_checkpoint_method,
+            activations_checkpoint_num_layers=activations_checkpoint_num_layers,
+            layernorm_epsilon=layernorm_epsilon,
+            bias_gelu_fusion=bias_gelu_fusion,
+            persist_layer_norm=persist_layer_norm,
+            openai_gelu=openai_gelu,
+            onnx_safe=onnx_safe,
+            hidden_steps=hidden_steps,
+            hidden_blocks=hidden_steps,
```

Comment:
Same bug, should be: `hidden_blocks=hidden_blocks`

Reply:
Good catch.Fixed.